sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusAsyncConsumer.java

```diff
@@ -26,17 +30,32 @@
     private final String linkName;
     private final ServiceBusReceiveLinkProcessor linkProcessor;
     private final MessageSerializer messageSerializer;
-    private final Flux<ServiceBusReceivedMessage> processor;
+    private final Flux<ServiceBusReceivedMessage> messageSource;
 
     ServiceBusAsyncConsumer(String linkName, ServiceBusReceiveLinkProcessor linkProcessor,
-        MessageSerializer messageSerializer, int prefetch) {
+        MessageSerializer messageSerializer, ReceiverOptions receiverOptions,
+        LockContainer<LockRenewalOperation> messageLockContainer, Function<String, Mono<OffsetDateTime>> onRenewLock) {
         this.linkName = linkName;
         this.linkProcessor = linkProcessor;
         this.messageSerializer = messageSerializer;
-        this.processor = linkProcessor
+
+        Flux<ServiceBusReceivedMessage> source = linkProcessor
             .map(message -> this.messageSerializer.deserialize(message, ServiceBusReceivedMessage.class))
-            .publish(prefetch)
+            .publish(receiverOptions.getPrefetchCount())
             .autoConnect(1);
+
+        final Duration maxAutoLockRenewDuration = receiverOptions.getMaxLockRenewDuration();
+
+        final boolean isAutoLockRenewal = maxAutoLockRenewDuration != null
```

Comment:
Can we move this duration logic into a receiver option? 
`boolean isAutoRenewEnabled() { return maxAutoLockRenewDuration != null ...; }`